packages/gatsby-source-contentful/src/__tests__/gatsby-node.js

```diff
@@ -0,0 +1,346 @@
+      getCache,
+    })
+
+    const newBlogPost = getNode(newBlogPostId)
+    const newBlogPostNl = getNode(newBlogPostIdNl)
+    // const author = getNode(newBlogPost[`author___NODE`])
+    // const authorNl = getNode(newBlogPostNl[`author___NODE`])
+
+    const expectedObjectUS = {
+      contentful_id: blogPostEntry.sys.id.substr(1),
+      title: blogPostEntry.fields.title[`en-US`],
+      slug: blogPostEntry.fields.slug[`en-US`],
+      node_locale: `en-US`,
+      description___NODE: createNodeId(`${newBlogPostId}descriptionTextNode`),
+      body___NODE: createNodeId(`${newBlogPostId}bodyTextNode`),
+      heroImage___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.heroImage[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `en-US`,
+      }),
+      author___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `en-US`,
+      }),
+    }
+    const expectedObjectNl = {
+      ...expectedObjectUS,
+      node_locale: `nl`,
+      description___NODE: createNodeId(`${newBlogPostIdNl}descriptionTextNode`),
+      body___NODE: createNodeId(`${newBlogPostIdNl}bodyTextNode`),
+      heroImage___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.heroImage[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `nl`,
+      }),
+      author___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `nl`,
+      }),
+    }
+
+    expect(newBlogPost).toMatchObject(expectedObjectUS)
+    expect(newBlogPostNl).toMatchObject(expectedObjectNl)
+
+    // test author links
+    // TODO fix
+    // expect(author[`blog post___NODE`]).toContain(newBlogPost.id)
+    // expect(authorNl[`blog post___NODE`]).toContain(newBlogPostNl.id)
+  })
+
+  it(`should update a blogpost`, async () => {
+    const updatedBlogPostId = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP`
+    const updatedBlogPostIdNl = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP___nl`
+    const blogPostEntry =
+      startersBlogFixture.updateBlogPost.currentSyncData.entries[0]
+
+    fetch
+      .mockReturnValueOnce(startersBlogFixture.initialSync)
+      .mockReturnValueOnce(startersBlogFixture.createBlogPost)
+      .mockReturnValueOnce(startersBlogFixture.updateBlogPost)
+
+    // initial sync
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    // create blog post
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
```

Comment:
similar to previous test - worth adding assertion that before updating blog post the title is different than final one?

Reply:
Is it necessary? cause the previous test shows that this already works

Reply:
It's similar but not exactly - the previous one checks if node was created, and this one would check if node was updated. It's not strictly necessary - up to you - feel free to resolve it without changes

Reply:
I will leave that decision to you - in previous one you check if node exists before -> add new post -> assert that new blog post node exists

in this if anything I would assert title